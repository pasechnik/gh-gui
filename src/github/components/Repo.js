import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ListGroupItem, ListGroupItemHeading, ListGroupItemText } from 'reactstrap'

import '../styles/repo.css'

class Repo extends Component {
  state = {
    visible: false,
  }

  changeVisible = () => {
    const { visible } = this.state

    this.setState({ visible: !visible })
  }

  render() {
    const { visible } = this.state
    const {
      htmlUrl, name, fork, forks, homepage, language, description, stargazersCount,
    } = this.props

    return (
      <ListGroupItem>
        <ListGroupItemHeading onClick={this.changeVisible} className='listHeading'>
          {name}
        </ListGroupItemHeading>
        <ListGroupItemText style={{ display: (visible === true ? '' : 'none') }} className='listBody'>
          <a href={htmlUrl} target='_blank' rel='noopener noreferrer'>
            {name}
          </a>
          <br />
          <span className='fork'>{fork === true ? 'it\'s a fork' : ''}</span>
          <br />
          <span className='forks'>
            Forks:
            {' '}
            {forks}
          </span>
          <br />
          <span className='stars'>
            Stars:
            {' '}
            {stargazersCount}
          </span>
          <br />
          <span className='language'>
            Language:
            {' '}
            {language}
          </span>
          <br />
          <span className='homepage'>
            Homepage:
            {' '}
            {homepage}
          </span>
          <br />
          <span className='description'>
            Description:
            {' '}
            {description}
          </span>
        </ListGroupItemText>
      </ListGroupItem>
    )
  }
}

Repo.propTypes = {
  htmlUrl: PropTypes.string,
  name: PropTypes.string,
  fork: PropTypes.bool,
  forks: PropTypes.number,
  homepage: PropTypes.string,
  language: PropTypes.string,
  description: PropTypes.string,
  stargazersCount: PropTypes.number,
}

Repo.defaultProps = {
  homepage: '',
  htmlUrl: '',
  name: '',
  fork: false,
  forks: 0,
  language: '',
  description: '',
  stargazersCount: 0,
}
export default Repo
