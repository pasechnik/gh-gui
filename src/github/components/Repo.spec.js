import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import renderer from 'react-test-renderer'

// import React from 'react'
// import { shallow, mount } from 'enzyme'
import Repo from './Repo'

configure({ adapter: new Adapter() })

const repo = {
  id: 11111,
  homepage: 'homepage',
  htmlUrl: 'html',
  name: 'name',
  fork: true,
  forks: 100,
  language: 'JS',
  description: 'description',
  stargazersCount: 20,
}

describe('<Repo />', () => {
  it('Renders empty Repo', () => {
    const component = renderer.create(<Repo />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Repo with props', () => {
    const component = renderer.create(<Repo key={repo.id} {...repo} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('<Repo /> shallow tests', () => {
  const component = shallow(<Repo key={repo.id} {...repo} />)

  it('Renders with props', () => {
    expect(component).toMatchSnapshot()
  })

  it('has ListGroupItem', () => {
    expect(component.find('ListGroupItem').length).toBe(1)
  })
  it('has ListGroupItemHeading', () => {
    expect(component.find('ListGroupItemHeading').length).toBe(1)
  })
  it('has a', () => {
    expect(component.find('a').length).toBe(1)
  })
  it('a has href=html', () => {
    expect(component.find('a').prop('href')).toEqual('html')
  })
  it('has a > name', () => {
    expect(component.find('a').text()).toEqual('name')
  })
  it('has span.fork', () => {
    expect(component.find('span.fork').length).toBe(1)
  })
  it('span.fork has text it\'s a fork', () => {
    expect(component.find('span.fork').text()).toEqual('it\'s a fork')
  })
  it('has span.forks', () => {
    expect(component.find('span.forks').length).toBe(1)
  })
  it('span.forks has text Forks: 100', () => {
    expect(component.find('span.forks').text()).toEqual('Forks: 100')
  })
  it('has span.stars', () => {
    expect(component.find('span.stars').length).toBe(1)
  })
  it('span.stars has text Stars: 20', () => {
    expect(component.find('span.stars').text()).toEqual('Stars: 20')
  })
  it('has span.language', () => {
    expect(component.find('span.language').length).toBe(1)
  })
  it('span.language has text Language: JS', () => {
    expect(component.find('span.language').text()).toEqual('Language: JS')
  })
  it('has span.homepage', () => {
    expect(component.find('span.homepage').length).toBe(1)
  })
  it('span.homepage has text Stars: 20', () => {
    expect(component.find('span.homepage').text()).toEqual('Homepage: homepage')
  })
  it('has span.description', () => {
    expect(component.find('span.description').length).toBe(1)
  })
  it('span.description has text Description: description', () => {
    expect(component.find('span.description').text()).toEqual('Description: description')
  })
})
