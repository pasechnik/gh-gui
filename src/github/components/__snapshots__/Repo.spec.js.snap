// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repo /> Renders Repo with props 1`] = `
<li
  className="list-group-item"
>
  <h5
    className="listHeading list-group-item-heading"
    onClick={[Function]}
  >
    name
  </h5>
  <p
    className="listBody list-group-item-text"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <a
      href="html"
      rel="noopener noreferrer"
      target="_blank"
    >
      name
    </a>
    <br />
    <span
      className="fork"
    >
      it's a fork
    </span>
    <br />
    <span
      className="forks"
    >
      Forks:
       
      100
    </span>
    <br />
    <span
      className="stars"
    >
      Stars:
       
      20
    </span>
    <br />
    <span
      className="language"
    >
      Language:
       
      JS
    </span>
    <br />
    <span
      className="homepage"
    >
      Homepage:
       
      homepage
    </span>
    <br />
    <span
      className="description"
    >
      Description:
       
      description
    </span>
  </p>
</li>
`;

exports[`<Repo /> Renders empty Repo 1`] = `
<li
  className="list-group-item"
>
  <h5
    className="listHeading list-group-item-heading"
    onClick={[Function]}
  >
    
  </h5>
  <p
    className="listBody list-group-item-text"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      
    </a>
    <br />
    <span
      className="fork"
    >
      
    </span>
    <br />
    <span
      className="forks"
    >
      Forks:
       
      0
    </span>
    <br />
    <span
      className="stars"
    >
      Stars:
       
      0
    </span>
    <br />
    <span
      className="language"
    >
      Language:
       
      
    </span>
    <br />
    <span
      className="homepage"
    >
      Homepage:
       
      
    </span>
    <br />
    <span
      className="description"
    >
      Description:
       
      
    </span>
  </p>
</li>
`;

exports[`<Repo /> shallow tests Renders with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Repo
    description="description"
    fork={true}
    forks={100}
    homepage="homepage"
    htmlUrl="html"
    id={11111}
    language="JS"
    name="name"
    stargazersCount={20}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListGroupItemHeading
          className="listHeading"
          onClick={[Function]}
          tag="h5"
>
          name
</ListGroupItemHeading>,
        <ListGroupItemText
          className="listBody"
          style={
                    Object {
                              "display": "none",
                            }
          }
          tag="p"
>
          <a
                    href="html"
                    rel="noopener noreferrer"
                    target="_blank"
          >
                    name
          </a>
          <br />
          <span
                    className="fork"
          >
                    it's a fork
          </span>
          <br />
          <span
                    className="forks"
          >
                    Forks:
                     
                    100
          </span>
          <br />
          <span
                    className="stars"
          >
                    Stars:
                     
                    20
          </span>
          <br />
          <span
                    className="language"
          >
                    Language:
                     
                    JS
          </span>
          <br />
          <span
                    className="homepage"
          >
                    Homepage:
                     
                    homepage
          </span>
          <br />
          <span
                    className="description"
          >
                    Description:
                     
                    description
          </span>
</ListGroupItemText>,
      ],
      "tag": "li",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "name",
          "className": "listHeading",
          "onClick": [Function],
          "tag": "h5",
        },
        "ref": null,
        "rendered": "name",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <a
              href="html"
              rel="noopener noreferrer"
              target="_blank"
>
              name
</a>,
            <br />,
            <span
              className="fork"
>
              it's a fork
</span>,
            <br />,
            <span
              className="forks"
>
              Forks:
               
              100
</span>,
            <br />,
            <span
              className="stars"
>
              Stars:
               
              20
</span>,
            <br />,
            <span
              className="language"
>
              Language:
               
              JS
</span>,
            <br />,
            <span
              className="homepage"
>
              Homepage:
               
              homepage
</span>,
            <br />,
            <span
              className="description"
>
              Description:
               
              description
</span>,
          ],
          "className": "listBody",
          "style": Object {
            "display": "none",
          },
          "tag": "p",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "name",
              "href": "html",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "name",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "it's a fork",
              "className": "fork",
            },
            "ref": null,
            "rendered": "it's a fork",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Forks:",
                " ",
                100,
              ],
              "className": "forks",
            },
            "ref": null,
            "rendered": Array [
              "Forks:",
              " ",
              100,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Stars:",
                " ",
                20,
              ],
              "className": "stars",
            },
            "ref": null,
            "rendered": Array [
              "Stars:",
              " ",
              20,
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Language:",
                " ",
                "JS",
              ],
              "className": "language",
            },
            "ref": null,
            "rendered": Array [
              "Language:",
              " ",
              "JS",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homepage:",
                " ",
                "homepage",
              ],
              "className": "homepage",
            },
            "ref": null,
            "rendered": Array [
              "Homepage:",
              " ",
              "homepage",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Description:",
                " ",
                "description",
              ],
              "className": "description",
            },
            "ref": null,
            "rendered": Array [
              "Description:",
              " ",
              "description",
            ],
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListGroupItemHeading
            className="listHeading"
            onClick={[Function]}
            tag="h5"
>
            name
</ListGroupItemHeading>,
          <ListGroupItemText
            className="listBody"
            style={
                        Object {
                                    "display": "none",
                                  }
            }
            tag="p"
>
            <a
                        href="html"
                        rel="noopener noreferrer"
                        target="_blank"
            >
                        name
            </a>
            <br />
            <span
                        className="fork"
            >
                        it's a fork
            </span>
            <br />
            <span
                        className="forks"
            >
                        Forks:
                         
                        100
            </span>
            <br />
            <span
                        className="stars"
            >
                        Stars:
                         
                        20
            </span>
            <br />
            <span
                        className="language"
            >
                        Language:
                         
                        JS
            </span>
            <br />
            <span
                        className="homepage"
            >
                        Homepage:
                         
                        homepage
            </span>
            <br />
            <span
                        className="description"
            >
                        Description:
                         
                        description
            </span>
</ListGroupItemText>,
        ],
        "tag": "li",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "name",
            "className": "listHeading",
            "onClick": [Function],
            "tag": "h5",
          },
          "ref": null,
          "rendered": "name",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <a
                href="html"
                rel="noopener noreferrer"
                target="_blank"
>
                name
</a>,
              <br />,
              <span
                className="fork"
>
                it's a fork
</span>,
              <br />,
              <span
                className="forks"
>
                Forks:
                 
                100
</span>,
              <br />,
              <span
                className="stars"
>
                Stars:
                 
                20
</span>,
              <br />,
              <span
                className="language"
>
                Language:
                 
                JS
</span>,
              <br />,
              <span
                className="homepage"
>
                Homepage:
                 
                homepage
</span>,
              <br />,
              <span
                className="description"
>
                Description:
                 
                description
</span>,
            ],
            "className": "listBody",
            "style": Object {
              "display": "none",
            },
            "tag": "p",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "name",
                "href": "html",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "name",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "it's a fork",
                "className": "fork",
              },
              "ref": null,
              "rendered": "it's a fork",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Forks:",
                  " ",
                  100,
                ],
                "className": "forks",
              },
              "ref": null,
              "rendered": Array [
                "Forks:",
                " ",
                100,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Stars:",
                  " ",
                  20,
                ],
                "className": "stars",
              },
              "ref": null,
              "rendered": Array [
                "Stars:",
                " ",
                20,
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Language:",
                  " ",
                  "JS",
                ],
                "className": "language",
              },
              "ref": null,
              "rendered": Array [
                "Language:",
                " ",
                "JS",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homepage:",
                  " ",
                  "homepage",
                ],
                "className": "homepage",
              },
              "ref": null,
              "rendered": Array [
                "Homepage:",
                " ",
                "homepage",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Description:",
                  " ",
                  "description",
                ],
                "className": "description",
              },
              "ref": null,
              "rendered": Array [
                "Description:",
                " ",
                "description",
              ],
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
